src/porepy/fracs/mortars.py

```diff
@@ -233,35 +240,36 @@ def match_grids_1d(new_1d, old_1d, tol):
               grid.
 
     """
+    cell_nodes1 = new_1d.cell_nodes()
+    cell_nodes2 = old_1d.cell_nodes()
+    nodes1 = pp.utils.mcolon.mcolon(cell_nodes1.indptr[0:-1], cell_nodes1.indptr[1:])
+    nodes2 = pp.utils.mcolon.mcolon(cell_nodes2.indptr[0:-1], cell_nodes2.indptr[1:])
 
-    # Create a grid that contains all nodes of both the old and new grids.
-    combined, _, new_ind, old_ind, _, _ = non_conforming.merge_1d_grids(
-        new_1d, old_1d, tol=tol
-    )
-    combined.compute_geometry()
-    weights = combined.cell_volumes
-
-    switch_new = new_ind[0] > new_ind[-1]
-    if switch_new:
-        new_ind = new_ind[::-1]
-    switch_old = old_ind[0] > old_ind[-1]
-    if switch_old:
-        old_ind = old_ind[::-1]
-
-    diff_new = np.diff(new_ind)
-    diff_old = np.diff(old_ind)
-    new_in_full = rldecode(np.arange(diff_new.size), diff_new)
-    old_in_full = rldecode(np.arange(diff_old.size), diff_old)
```

Comment:
I replace 100% of this function. I belive the issue might partly be in these two lines. This seems to assume that the the nodes and cells are ordered as given from pp.TensorGrid. Also, I belive there is some issues with non_conforming.merge_1d_grids(..) when the grid has two nodes at the same point (i.e. a 0d intersection).

I believe the new code can handle this